modules/vmfcore/java/com/intel/vmf/FieldDesc.java

```diff
@@ -0,0 +1,84 @@
+package com.intel.vmf;
+
+public class FieldDesc
+{
+    static
+    {
+        System.loadLibrary("vmf");
+    }
+	
+    protected final long nativeObj;
+    
+    protected FieldDesc (long addr)
+    {
+        if (addr == 0)
+            throw new java.lang.UnsupportedOperationException("Native object address is NULL");
+        
+        nativeObj = addr;
+    }
+    
+    public FieldDesc ()
+    {
+        nativeObj = n_FieldDesc ();
+    }
+	
+    public FieldDesc (String newName, int newType, boolean isOptional)
```

Comment:
I suggest one more c-tor: `FieldDesc (String name, int type)` setting `isOptional = false` - that probably will be the most used.